geofileops/util/_geoops_sql.py

```diff
@@ -2121,14 +2138,17 @@ def join_by_location(
     )
 
 
-def _prepare_filter_by_location_fields(
+def _add_specific_optimisation(
```

Comment:
I think it would be useful to write some specific, simple tests on this function, e.g. to check for some queries trhat the spatial_relation_filter returned is wat is to be expected. Now there are only high-level tests where it is much harder to test "special cases" or e.g. to test for "unsupported" things, errors,...